CRM/Contact/BAO/Contact.php

```diff
@@ -2599,7 +2599,11 @@ public static function getValues(&$params, &$values) {
       if ($contact->birth_date) {
         $birthDate = CRM_Utils_Date::customFormat($contact->birth_date, '%Y%m%d');
         if ($birthDate < date('Ymd')) {
-          $age = CRM_Utils_Date::calculateAge($birthDate);
+          $deceasedDate = NULL;
+          if (!empty($contact->is_deceased) && !empty($contact->deceased_date)) {
+            $deceasedDate = $contact->deceased_date;
+          }
+          $age = CRM_Utils_Date::calculateAge($birthDate, $deceasedDate);
```

Comment:
You could simplify this to 
```
CRM_Utils_Date::calculateAge($birthDate, $contact->deceased_date ?? NULL);
```

Reply:
@eileenmcnaughton 
My changes were correct, if contact is marked `is_deceased` with `deceased_date`. Then unset the flag, `deceased_date` date remain present in DB even we unset the date in the form.

In this situation, we need to check both
`if (!empty($contact->is_deceased) && !empty($contact->deceased_date)) {`

Reply:
```php
    [birth_date] => 1961-06-06
    [is_deceased] => 0
    [deceased_date] => 1991-07-07

```